cmd/util/flags/cliflags/repo.go

```diff
@@ -0,0 +1,65 @@
+package cliflags
+
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/rs/zerolog/log"
+)
+
+const defaultBacalhauDir = ".bacalhau"
+
+// getDefaultRepo determines the appropriate default directory for storing repository data.
+// Priority order:
+// 1. If the environment variable BACALHAU_DIR is set and non-empty, use it.
+// 2. If the environment variable FIL_WALLET_ADDRESS is set, it tries to use ROOT_DIR.
+// 3. User's home directory with .bacalhau appended.
+// 4. User-specific configuration directory with .bacalhau appended.
+// 5. If all above fail, use .bacalhau in the current directory.
+// The function logs any errors encountered during the process and always returns a usable path.
+func getDefaultRepo() string {
+	if repoDir, set := os.LookupEnv("BACALHAU_DIR"); set && repoDir != "" {
```

Comment:
I believe we will still require a method like this in the event a config file isn't provided, or if one is provided without the `Repo` field set.

Reply:
even we decide on a default config path, the mental model should be:
1. look for a config flag and load the configuration from the user provided config file, if provided
2. _iff_ no config file was provided, then check for a config file in the default path (e.g. `~/.config/bacalhau/node_config.yaml`), and load the configuration. By default, this config file should be empty
3. For every config value, we should replace empty values with default values that are hardcoded or resolved at runtime.

I am trying to avoid forking the logic or treating the DataDir different than other configurations. If no DataDir value is provided in the config file or config flags, then we use its default value the same was as we use default values for any other config

Reply:
> iff no config file was provided, then check for a config file in the default path (e.g. ~/.config/bacalhau/node_config.yaml), and load the configuration. By default, this config file should be empty

This is the current behavior. And if one isn't present there we look in the repo.